### PR TITLE
Fix Detox Tests

### PR DESCRIPTION
**Pull request by utkarsh-dixit**
## Summary

Detox tests were failing because of `DisplayIfVisible` component and `YellowBox`. This PR is meant to fix these issues

## Changelog

[General] [Removed]: Remove DisplayIfVisible component and use simple if else conditions to show which component to show.
[General] [Fixed]: Disable Yellow box so that it doesn't block detox to perform any action on api's tab at bottom.
[General] [Changed]: Update detox from 16.7.2 to 17.10.0
[General][Removed]: Exclude arm64 for rn-tester iOS project
[General][Changed]: Updated pod file for cocoapods dependencies

## Test Plan
You can test this PR by,

`yarn build-ios-e2e && yarn test-ios-e2e`

All Test Passing

### CODE REVIEW

**Comment by hramos** [comment]
Thanks for the fix! Can you re-enable the detox tests on Circle CI as part of this PR? I think you'd only need to uncomment some lines in `.circleci/config.yml`, and then we'll be able to confirm these are fixed.

**Comment by utkarsh-dixit** [comment]
@hramos I enabled detox tests for iOS in circleci but they are failing because react native fails to build for iOS (mainly for arm64 and i386). I faced the same issue locally when I was building rn-tester for iOS, which I fixed by following the solution in this issue #29984, which was to exclude `arm64` and `i386` architectures when build rn tester app. 


`ld: in /Users/distiller/react-native/packages/rn-tester/Pods/OpenSSL-Universal/ios/lib/libcrypto.a(cryptlib.o), building for iOS Simulator, but linking in object file built for iOS, file '/Users/distiller/react-native/packages/rn-tester/Pods/OpenSSL-Universal/ios/lib/libcrypto.a' for architecture arm64`

Do you have any ideas about what is causing this issue?

**Comment by mkcode** [comment]
There are a few issues here:
 * There is a compilation error of fsevents during the yarn install step
 * We need to `pod update` in order to get the latest Flipper-Folly/OpenSSLUniversal to solve this issue: https://github.com/facebook/react-native/issues/28824
 * We need to disable building for arm64 for now. See: https://stackoverflow.com/questions/63607158/xcode-12-building-for-ios-simulator-but-linking-in-object-file-built-for-ios

**Comment by utkarsh-dixit** [comment]
@hramos Can you review this PR again, the tests are successfully passing on circle CI now

**Comment by mkcode** [comment]
@utkarsh-dixit - Let's fix the analyze code step first.

File: packages/rn-tester/js/RNTesterAppShared.js [hunk 1]
```diff
@@ -45,11 +45,9 @@ const APP_STATE_KEY = 'RNTesterAppState.v3';
...
 LogBox.ignoreLogs([/AsyncStorage has been extracted from react-native/]);
 
-const DisplayIfVisible = ({isVisible, children}) => (
-  <View style={[styles.container, !isVisible && styles.hidden]}>
-    {children}
-  </View>
-);
+// Disable yellowbox so that they don't cover the API's tab and make it
+// unaccessible for detox.
+LogBox.ignoreAllLogs();
```
Comment (mkcode) [inline packages/rn-tester/js/RNTesterAppShared.js:50]: Let's test for jest presence here and only disable Logs in detox where jest is defined.

**Review by mkcode — CHANGES_REQUESTED** [review]

**Comment by kelset** [comment]
closing this since Detox is not in the repo anymore: https://github.com/facebook/react-native/pull/32907

**Pull request closed without merging**